### ISSUE TITLE
Read time out on uploading a large collection with mmap

### ISSUE THREAD

**Issue by eu9ene** [qdrant/qdrant#1825]
## Current Behavior
I'm uploading a large number of vectors to mmap collection with disabled indexing and getting `timeout: The read operation timed out`. I managed to upload 25667264 160-dimensional vectors, then it started responding with timeouts.

The idea is to use such a collection for vector lookup in `recommend` method.

## Steps to Reproduce
1. Create a mmap collection with disabled indexing

```
client.recreate_collection(
    collection_name=COLLECTION_ALL_MMAP,
    vectors_config=models.VectorParams(size=160, distance=models.Distance.DOT),
    optimizers_config=models.OptimizersConfigDiff(
        indexing_threshold=1000000000,
        memmap_threshold=1000
    )
)
   
```

2. Upload random vectors in batches:
```
def upload(collection, num_vectors):
    partition_size = min(1000000, num_vectors)

    while True:
        try:
            start = client.count(collection).count

            if start == num_vectors:
                break

            vectors = create_random_vectors(partition_size)
            ids = list(range(start, start+partition_size))

            print(f'new partition {ids[0]}, {ids[-1]}')

            client.upload_collection(
                    collection_name=collection,
                    vectors=vectors,
                    payload=None,
                    ids=ids,  
                    parallel=16
            )
        except Exception as ex:
            print(ex)
            time.sleep(1)
            print('retrying')

upload(COLLECTION_ALL_MMAP, 100000000)  
```

## Expected Behavior
The vectors are uploading correctly and all the required extra batching is handled inside the `upload_collection` method.
Even if EFS is slower, it doesn't timeout the uploading.

## Actual

It uploads a part of vectors, then any further uploading fails with the exception:
```
---------------------------------------------------------------------------
timeout                                   Traceback (most recent call last)
~/opt/anaconda3/lib/python3.8/site-packages/httpcore/_exceptions.py in map_exceptions(map)
      7     try:
----> 8         yield
      9     except Exception as exc:  # noqa: PIE786

~/opt/anaconda3/lib/python3.8/site-packages/httpcore/backends/sync.py in read(self, max_bytes, timeout)
     25             self._sock.settimeout(timeout)
---> 26             return self._sock.recv(max_bytes)
     27 

~/opt/anaconda3/lib/python3.8/ssl.py in recv(self, buflen, flags)
   1225                     self.__class__)
-> 1226             return self.read(buflen)
   1227         else:

~/opt/anaconda3/lib/python3.8/ssl.py in read(self, len, buffer)
   1100             else:
-> 1101                 return self._sslobj.read(len)
   1102         except SSLError as x:

timeout: The read operation timed out

During handling of the above exception, another exception occurred:

ReadTimeout                               Traceback (most recent call last)
~/opt/anaconda3/lib/python3.8/site-packages/httpx/_transports/default.py in map_httpcore_exceptions()
     59     try:
---> 60         yield
     61     except Exception as exc:  # noqa: PIE-786

~/opt/anaconda3/lib/python3.8/site-packages/httpx/_transports/default.py in handle_request(self, request)
    217         with map_httpcore_exceptions():
--> 218             resp = self._pool.handle_request(req)
    219 

~/opt/anaconda3/lib/python3.8/site-packages/httpcore/_sync/connection_pool.py in handle_request(self, request)
    252                 self.response_closed(status)
--> 253                 raise exc
    254             else:

~/opt/anaconda3/lib/python3.8/site-packages/httpcore/_sync/connection_pool.py in handle_request(self, request)
    236             try:
--> 237                 response = connection.handle_request(request)
    238             except ConnectionNotAvailable:

~/opt/anaconda3/lib/python3.8/site-packages/httpcore/_sync/connection.py in handle_request(self, request)
     89 
---> 90         return self._connection.handle_request(request)
     91 

~/opt/anaconda3/lib/python3.8/site-packages/httpcore/_sync/http11.py in handle_request(self, request)
    104                 self._response_closed()
--> 105             raise exc
    106 

~/opt/anaconda3/lib/python3.8/site-packages/httpcore/_sync/http11.py in handle_request(self, request)
     83                     headers,
---> 84                 ) = self._receive_response_headers(**kwargs)
     85                 trace.return_value = (

~/opt/anaconda3/lib/python3.8/site-packages/httpcore/_sync/http11.py in _receive_response_headers(self, request)
    147         while True:
--> 148             event = self._receive_event(timeout=timeout)
    149             if isinstance(event, h11.Response):

~/opt/anaconda3/lib/python3.8/site-packages/httpcore/_sync/http11.py in _receive_event(self, timeout)
    176             if event is h11.NEED_DATA:
--> 177                 data = self._network_stream.read(
    178                     self.READ_NUM_BYTES, timeout=timeout

~/opt/anaconda3/lib/python3.8/site-packages/httpcore/backends/sync.py in read(self, max_bytes, timeout)
     25             self._sock.settimeout(timeout)
---> 26             return self._sock.recv(max_bytes)
     27 

~/opt/anaconda3/lib/python3.8/contextlib.py in __exit__(self, type, value, traceback)
    130             try:
--> 131                 self.gen.throw(type, value, traceback)
    132             except StopIteration as exc:

~/opt/anaconda3/lib/python3.8/site-packages/httpcore/_exceptions.py in map_exceptions(map)
     11             if isinstance(exc, from_exc):
---> 12                 raise to_exc(exc)
     13         raise  # pragma: nocover

ReadTimeout: The read operation timed out

The above exception was the direct cause of the following exception:

ReadTimeout                               Traceback (most recent call last)
~/opt/anaconda3/lib/python3.8/site-packages/qdrant_client/http/api_client.py in send_inner(self, request)
     94         try:
---> 95             response = self._client.send(request)
     96         except Exception as e:

~/opt/anaconda3/lib/python3.8/site-packages/httpx/_client.py in send(self, request, stream, auth, follow_redirects)
    901 
--> 902         response = self._send_handling_auth(
    903             request,

~/opt/anaconda3/lib/python3.8/site-packages/httpx/_client.py in _send_handling_auth(self, request, auth, follow_redirects, history)
    929             while True:
--> 930                 response = self._send_handling_redirects(
    931                     request,

~/opt/anaconda3/lib/python3.8/site-packages/httpx/_client.py in _send_handling_redirects(self, request, follow_redirects, history)
    966 
--> 967             response = self._send_single_request(request)
    968             try:

~/opt/anaconda3/lib/python3.8/site-packages/httpx/_client.py in _send_single_request(self, request)
   1002         with request_context(request=request):
-> 1003             response = transport.handle_request(request)
   1004 

~/opt/anaconda3/lib/python3.8/site-packages/httpx/_transports/default.py in handle_request(self, request)
    217         with map_httpcore_exceptions():
--> 218             resp = self._pool.handle_request(req)
    219 

~/opt/anaconda3/lib/python3.8/contextlib.py in __exit__(self, type, value, traceback)
    130             try:
--> 131                 self.gen.throw(type, value, traceback)
    132             except StopIteration as exc:

~/opt/anaconda3/lib/python3.8/site-packages/httpx/_transports/default.py in map_httpcore_exceptions()
     76         message = str(exc)
---> 77         raise mapped_exc(message) from exc
     78 

ReadTimeout: The read operation timed out

During handling of the above exception, another exception occurred:

ResponseHandlingException                 Traceback (most recent call last)
<ipython-input-79-679fbd791b1f> in <module>
----> 1 upload(COLLECTION_ALL_MMAP, NUM_VECTORS_ALL)

<ipython-input-77-c5780c7073b2> in upload(collection, num_vectors, payload)
      5 #     while True:
      6 #         try:
----> 7     start = client.count(collection).count
      8 
      9 #     if start == num_vectors:

~/opt/anaconda3/lib/python3.8/site-packages/qdrant_client/qdrant_client.py in count(self, collection_name, count_filter, exact)
    674             count_filter = GrpcToRest.convert_filter(model=count_filter)
    675 
--> 676         count_result = self.openapi_client.points_api.count_points(
    677             collection_name=collection_name,
    678             count_request=rest.CountRequest(

~/opt/anaconda3/lib/python3.8/site-packages/qdrant_client/http/api/points_api.py in count_points(self, collection_name, count_request)
    717         Count points which matches given filtering condition
    718         """
--> 719         return self._build_for_count_points(
    720             collection_name=collection_name,
    721             count_request=count_request,

~/opt/anaconda3/lib/python3.8/site-packages/qdrant_client/http/api/points_api.py in _build_for_count_points(self, collection_name, count_request)
    196         body = jsonable_encoder(count_request)
    197 
--> 198         return self.api_client.request(
    199             type_=m.InlineResponse20015,
    200             method="POST",

~/opt/anaconda3/lib/python3.8/site-packages/qdrant_client/http/api_client.py in request(self, type_, method, url, path_params, **kwargs)
     66         url = (self.host or "") + url.format(**path_params)
     67         request = self._client.build_request(method, url, **kwargs)
---> 68         return self.send(request, type_)
     69 
     70     @overload

~/opt/anaconda3/lib/python3.8/site-packages/qdrant_client/http/api_client.py in send(self, request, type_)
     83 
     84     def send(self, request: Request, type_: Type[T]) -> T:
---> 85         response = self.middleware(request, self.send_inner)
     86         if response.status_code in [200, 201]:
     87             try:

~/opt/anaconda3/lib/python3.8/site-packages/qdrant_client/http/api_client.py in __call__(self, request, call_next)
    186 class BaseMiddleware:
    187     def __call__(self, request: Request, call_next: Send) -> Response:
--> 188         return call_next(request)
    189 
    190 

~/opt/anaconda3/lib/python3.8/site-packages/qdrant_client/http/api_client.py in send_inner(self, request)
     95             response = self._client.send(request)
     96         except Exception as e:
---> 97             raise ResponseHandlingException(e)
     98         return response
     99 

ResponseHandlingException: The read operation timed out
```

State of the collection:

```
{
  "result": {
    "status": "yellow",
    "optimizer_status": "ok",
    "vectors_count": 25667264,
    "indexed_vectors_count": 0,
    "points_count": 25667264,
    "segments_count": 8,
    "config": {
      "params": {
        "vectors": {
          "size": 160,
          "distance": "Dot"
        },
        "shard_number": 1,
        "replication_factor": 1,
        "write_consistency_factor": 1,
        "on_disk_payload": true
      },
      "hnsw_config": {
        "m": 16,
        "ef_construct": 100,
        "full_scan_threshold": 10000,
        "max_indexing_threads": 0,
        "on_disk": false
      },
      "optimizer_config": {
        "deleted_threshold": 0.2,
        "vacuum_min_vector_number": 1000,
        "default_segment_number": 0,
        "max_segment_size": null,
        "memmap_threshold": 1000,
        "indexing_threshold": 1000000000,
        "flush_interval_sec": 5,
        "max_optimization_threads": 1
      },
      "wal_config": {
        "wal_capacity_mb": 32,
        "wal_segments_ahead": 0
      },
      "quantization_config": null
    },
    "payload_schema": {}
  },
  "status": "ok",
  "time": 0.000027102
}
```

## Context (Environment)
Deployment is AWS ECS + Fargate + EFS, single instance, distributed mode disabled, 16 cores 32GB RAM
Qdrant version 1.1.2

**Comment by generall** [comment]
Hi @eu9ene!
EFS might be too slow for mmap option. Could you please check the number of IOPS your disks are able to maintain?

Qdrant does periodic flushes, with slow disks it might take longer than default timeout of 10 seconds.
You can also increase it in client settings: `QdrantClient(..., timeout=100)`

**Comment by eu9ene** [comment]
> EFS might be too slow for mmap option

That's what I suspected. I tried `QdrantClient(..., timeout=100)` and it works for a bit and then starts failing with `Broken pipe`. I'm not sure how to calculate the exact IOPS on EFS but it seems like it doesn't hit the limit.

<img width="1207" alt="Screenshot 2023-05-02 at 4 08 08 PM" src="https://user-images.githubusercontent.com/2486505/235806775-368efb31-3e1f-41db-94b6-5ecbb44d0a44.png">
<img width="1211" alt="Screenshot 2023-05-02 at 4 11 29 PM" src="https://user-images.githubusercontent.com/2486505/235806869-72c8850d-88d6-4002-8829-1e952e8ad81f.png">
